includes/admin/class-sensei-wccom-connect-notice.php

```diff
@@ -0,0 +1,226 @@
+<?php
+/**
+ * File containing the class Sensei_WCCOM_Connect_Notice
+ *
+ * @package sensei-lms
+ * @since   3.10.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class that handles showing the WooCommerce.com connect notice.
+ *
+ * @access private
+ *
+ * @class Sensei_WCCOM_Connect_Notice
+ */
+class Sensei_WCCOM_Connect_Notice {
+	const DISMISS_NOTICE_NONCE_ACTION   = 'sensei-lms-cancelled-wccom-connect-dismiss';
+	const DISMISSED_NOTIFICATION_OPTION = 'sensei-cancelled-wccom-connect-dismissed';
+	const SENSEI_WCCOM_EXTENSIONS       = [
+		'152116:bad2a02a063555b7e2bee59924690763', // WC Paid Courses.
+		'543363:8ee2cdf89f55727f57733133ccbbfbb0', // Content Drip.
+		'435830:700f6f6786c764debcd5dfb789f5f506', // Share your Grade.
+		'435834:f6479a8a3a01ac11794f32be22b0682f', // Course Participants.
+	];
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		add_filter( 'admin_notices', [ $this, 'add_admin_notice' ] );
+		add_action( 'wp_ajax_sensei_dismiss_wccom_connect_notice', [ $this, 'handle_notice_dismiss' ] );
+	}
+
+	/**
+	 * Check if WooCommerce.com connection has been made.
+	 *
+	 * @return bool
+	 */
+	private function is_wccom_connected() {
+		if ( ! class_exists( 'WC_Helper_Options' ) ) {
+			// We can't do anything until WooCommerce is activated. There is already a notice for that.
+			return true;
+		}
+
+		$auth = WC_Helper_Options::get( 'auth' );
+
+		return ! empty( $auth['access_token'] );
+	}
+
+	/**
+	 * Get the connect URL.
+	 */
+	private function get_wccom_connect_url() {
+		return add_query_arg(
+			[
+				'page'              => 'wc-addons',
+				'section'           => 'helper',
```

Comment:
This is where a lot of magic happens. We currently can't control the redirect URL directly, but we could set a `x` minute transient that hooks into `woocommerce_helper_connected` and redirects to a location we choose if someone connects within that time.

Reply:
It returns to the WC Extensions > Subscriptions page, which shows the Sensei extension subscriptions, I think that's pretty good, seeing that page clarifies how it's all connected.